endpoints/citrus-kafka/src/test/java/org/citrusframework/kafka/integration/KafkaEndpointJavaIT.java

```diff
@@ -0,0 +1,232 @@
+        );
+
+        when(sleep().seconds(2));
+
+        ThrowableAssert.ThrowingCallable receiver = () -> then(
+                receive(kafkaWithRandomConsumerGroupEndpoint)
+                        .selector(
+                                kafkaConsumerSelector()
+                                        .eventLookbackWindow(Duration.ofSeconds(1L))
+                                        .kafkaConsumerRecordMatcher(kafkaHeaderEquals(key, value))
+                                        .build()
+                        )
+                        .getMessageBuilderSupport()
+                        .body(body)
+        );
+
+        assertThatThrownBy(receiver)
+                .isInstanceOf(TestCaseFailedException.class)
+                .hasRootCauseInstanceOf(CitrusRuntimeException.class)
+                .hasMessageContaining("Failed to resolve Kafka message using selector");
+    }
+
+    @Test
+    @CitrusTest
+    public void findKafkaEvent_duplicateEntriesFound_citrus_DSL() {
+        var body = "findKafkaEvent_duplicateEntriesFound_citrus_DSL";
+
+        var key = "Name";
+
+        given(
+                send(kafkaWithRandomConsumerGroupEndpoint)
+                        .message(new KafkaMessage(body).setHeader(key, "Gandalf the Grey"))
+        );
+
+        when(
+                send(kafkaWithRandomConsumerGroupEndpoint)
+                        .message(new KafkaMessage(body).setHeader(key, "Gandalf the White"))
+        );
+
+        ThrowableAssert.ThrowingCallable receiver = () -> then(
+                receive(kafkaWithRandomConsumerGroupEndpoint)
+                        .selector(
+                                kafkaConsumerSelector()
+                                        .eventLookbackWindow(Duration.ofSeconds(1L))
+                                        .kafkaConsumerRecordMatcher(kafkaHeaderContains(key, "Gandalf"))
+                                        .build()
+                        )
+                        .getMessageBuilderSupport()
+                        .body(body)
+        );
+
+        assertThatThrownBy(receiver)
+                .isInstanceOf(TestCaseFailedException.class)
+                .hasRootCauseInstanceOf(CitrusRuntimeException.class)
+                .hasMessageContaining("More than one matching record found in topic");
+    }
+
+    @Test
+    @CitrusTest
+    public void findKafkaEvent_headerEquals_java_DSL() {
+        var body = "findKafkaEvent_headerEquals_java_DSL";
+
+        var key = "Name";
+        var value = "Gollum";
+
+        when(
+                send(kafkaWithRandomConsumerGroupEndpoint)
+                        .message(new KafkaMessage(body).setHeader(key, value))
+        );
+
+        then(
+                kafkaWithRandomConsumerGroupEndpoint.findKafkaEventHeaderEquals(Duration.ofSeconds(1L), key, value)
+                        .body(body)
+        );
+    }
+
+    @Configuration
```

Comment:
```suggestion
    @TestConfiguration
```

Reply:
that's strangely not available here. I can double-check, but it might need an additional module.

Reply:
yeah well, that's part of `spring-boot`. not sure if the additional dependency is worth it. wdym?

Reply:
update: there is even a strange context-loading problem when taking spring boot onto the classpath. I won't do it.